Content.Server/Emp/EmpSystem.cs

```diff
@@ -0,0 +1,39 @@
+using Content.Server.Explosion.EntitySystems;
+using Content.Server.Light.Components;
+using Content.Server.Light.EntitySystems;
+using Content.Server.Power.Components;
+using Robust.Shared.Map;
+
+namespace Content.Server.Emp;
+
+public sealed class EmpSystem : EntitySystem
+{
+    [Dependency] private readonly EntityLookupSystem _lookup = default!;
+    [Dependency] private readonly PoweredLightSystem _poweredLight = default!;
+
+    public const string EmpPulseEffectPrototype = "EffectEmpPulse";
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<EmpOnTriggerComponent, TriggerEvent>(HandleEmpTrigger);
+    }
+
+    public void EmpPulse(EntityCoordinates coordinates, float range, float energyConsumption)
+    {
+        foreach (var uid in _lookup.GetEntitiesInRange(coordinates.ToMap(EntityManager), range))
```

Comment:
Doesn't this have an overload that takes entitycoordinates, also isn't ToMap obsolete.

Reply:
ToMap is not obsolete